master/buildbot/db/migrate/versions/057_add_build_data_tables.py

```diff
@@ -0,0 +1,49 @@
+# This file is part of Buildbot. Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+
+import sqlalchemy as sa
+
+from buildbot.util import sautils
+
+
+def upgrade(migrate_engine):
+    metadata = sa.MetaData()
+    metadata.bind = migrate_engine
+
+    sautils.Table(
+        'builds', metadata,
+        sa.Column('id', sa.Integer, primary_key=True),
+        # ...
+    )
+
+    build_data = sautils.Table(
+        'build_data', metadata,
+        sa.Column('id', sa.Integer, primary_key=True),
+        sa.Column('buildid', sa.Integer,
+                  sa.ForeignKey('builds.id', ondelete='CASCADE'),
+                  nullable=False),
+        sa.Column('name', sa.String(256), nullable=False),
+        # JSON encoded value
+        sa.Column('value_json', sa.Text(1000000), nullable=False),
```

Comment:
I would use Blob for storage as it is supposed to be stored outside of the table.

Reply:
also text is often stored in utf16 so it is double the needed size.

Reply:
Changed to Blob.